juju/juju
# 2.1 containerizer config 1657449

## Description of change

This exposes a method on Provider that indicates whether we think we can Allocate addresses for containers on that provider. It then uses that method to decide whether it will ask the containers to be provisioned on the local bridge, or whether it will ask to bridge them onto the host device.

## QA steps

With this change, you should be able to:
```
  $ juju bootstrap aws
  $ juju switch controller
  $ juju add-machine lxd:0
```
And have that container get properly provisioned.

## Bug reference

https://pad.lv/1657449
